packages/contracts/test/multi_sig_with_time_lock_except_remove_auth_addr.ts

```diff
@@ -0,0 +1,188 @@
+import { LogWithDecodedArgs, ZeroEx } from '0x.js';
+import { BlockchainLifecycle, devConstants, web3Factory } from '@0xproject/dev-utils';
+import { AbiDecoder } from '@0xproject/utils';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
+import * as chai from 'chai';
+import * as Web3 from 'web3';
+
+import { artifacts } from '../util/artifacts';
+import { constants } from '../util/constants';
+import { crypto } from '../util/crypto';
+import { MultiSigWrapper } from '../util/multi_sig_wrapper';
+import { SubmissionContractEventArgs, TransactionDataParams } from '../util/types';
+
+import { chaiSetup } from './utils/chai_setup';
+import { deployer } from './utils/deployer';
+const PROXY_ABI = artifacts.TokenTransferProxyArtifact.networks[constants.TESTRPC_NETWORK_ID].abi;
+const MUTISIG_WALLET_WITH_TIME_LOCK_EXCEPT_REMOVE_AUTHORIZED_ADDRESS_ABI =
+    artifacts.MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddressArtifact.networks[constants.TESTRPC_NETWORK_ID]
+        .abi;
+
+chaiSetup.configure();
+const expect = chai.expect;
+const web3 = web3Factory.create();
+const web3Wrapper = new Web3Wrapper(web3.currentProvider);
+const blockchainLifecycle = new BlockchainLifecycle(devConstants.RPC_URL);
+const abiDecoder = new AbiDecoder([MUTISIG_WALLET_WITH_TIME_LOCK_EXCEPT_REMOVE_AUTHORIZED_ADDRESS_ABI]);
+
+describe('MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress', () => {
+    const zeroEx = new ZeroEx(web3.currentProvider, { networkId: constants.TESTRPC_NETWORK_ID });
+    let owners: string[];
+    const requiredApprovals = 2;
+    const SECONDS_TIME_LOCKED = 1000000;
+
+    // initialize fake addresses
+    let authorizedAddress: string;
+    let unauthorizedAddress: string;
+
+    let tokenTransferProxy: Web3.ContractInstance;
+    let multiSig: Web3.ContractInstance;
+    let multiSigWrapper: MultiSigWrapper;
+
+    let validDestination: string;
+    before(async () => {
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        owners = [accounts[0], accounts[1]];
+        authorizedAddress = `0x${crypto
+            .solSHA3([accounts[0]])
+            .slice(0, 20)
+            .toString('hex')}`;
+        unauthorizedAddress = `0x${crypto
+            .solSHA3([accounts[1]])
+            .slice(0, 20)
+            .toString('hex')}`;
```

Comment:
These transformations should be in a helper function so that they aren't repeated. I know this is legacy code but if you could improve it, that would be awesome.